### ISSUE TITLE
App acquires location but doesn't send anything to my server if the app is terminated.

### ISSUE THREAD

**Issue by awesomeguy3** [transistorsoft/cordova-background-geolocation-lt#381]
## Your Environment
* Plugin version: latest
* Platform:  Android
* OS version: 7.0.1
* Device manufacturer / model: Samsung Galaxy S7
* Cordova version (`cordova -v`):
* Cordova platform version (`cordova platform ls`): 6.2.0
* Plugin config
```
BackgroundGeolocation.configure({
		// Geolocation config
		foregroundService: true,
		desiredAccuracy: 0,
		distanceFilter: 10,
		stationaryRadius: 25,
		// Activity Recognition config
		activityRecognitionInterval: 10000,
		stopTimeout: 5555,
		// Application config
		debug: true,  // <-- Debug sounds & notifications.
		stopOnTerminate: false,
		startOnBoot: true,
		heartbeatInterval: 30,
		// HTTP / SQLite config
		url: 'http://myserver.com/api/set_location',
		method: "POST",
		httpRootProperty: '.',
		locationTemplate: '{ "latitude":<%= latitude %>, "longitude":<%= longitude %> }',
		autoSync: true,
		maxDaysToPersist: 3,
		headers: {  // <-- Optional HTTP headers
			"X-FOO": "bar"
		},
		params: {   // <-- Optional HTTP params
			"auth_token": "maybe_your_server_authenticates_via_token_YES?",
			user_id: 1
		}
	}, function(state) {
		// This callback is executed when the plugin is ready to use.
		console.log("BackgroundGeolocation ready: ", state);
		if (!state.enabled) {
			BackgroundGeolocation.start();
		}
	});
```
## Expected Behavior
What I expect is the BackgroundLocation to send data to my server with it's http configuration and it doesn't do it at all. 
Also, I setup heartbeat and location events, they work fine while the app is running or if it is minimized.
When I terminate the app, it doesn't send data to my server although the icon in notification bar is present and debug sounds fire as expected.

## Actual Behavior
It doesn't send data on it's own. Hearbeat and location events send data only if the app is running, if it is closed (terminated), it still beeps like it's getting location and it is turning the gps icon on as usual, it just doesn't send any data to my server even though I used native XMLHttpRequest instead of jQuery (tried both), and even though I don't reference any variable from the app itself.

## Context
Trying to track the location of the device even if the app is closed (ofcourse with the consent of the user).

## Debug logs
There are no errors while the app is running, however, I have no idea how to track logs of it if I terminate the app.

**Comment by christocracy** [comment]
When an Android app is terminated, your Javascript **is terminated**.  If you're hoping to use Javascript to send data to your server after the Android app is terminated, that's **not** going to work.  Only the plugin's native Android service is running in this case.  The native service will continue to post locations to your configured `#url`.

See the docs for [stopOnTerminate](https://github.com/transistorsoft/cordova-background-geolocation-lt/tree/master/docs#config-boolean-stoponterminate-true)

**Comment by awesomeguy3** [comment]
As you can see from the post, stopOnTerminate is set to false, and as I have said already, the native service that is supposed to post the location to my server, doesn't do it at all.

**Comment by christocracy** [comment]
Samsung devices have particularly strict Battery Saving settings.

Check your Battery Saving settings.

**Comment by awesomeguy3** [comment]
Okay, I've done the following, disabled all battery saving settings, even tested on another device. 
Plugin's native service does make a request to my server, but it doesn't send any data at all. I've checked PHP's getallheaders(), $_REQUEST, $_SERVER, $_POST, $_GET and so on, none of them have any data related  to this plugin, not even parameters that I hardcoded it to send, nothing. 

I even tried just basic settings with only url and method set (for http request).
